File: General/MilleMercisTest/MilleMercisTest/PhotoViewController.m
//
//  PhotoViewController.m
//  MilleMercisTest
//
//  Created by Ancil on 9/9/15.
//  Copyright (c) 2015 Ancil Marshall. All rights reserved.
//

#import "PhotoViewController.h"
#import "AppDelegate.h"

@interface PhotoViewController ()
@property (weak, nonatomic) IBOutlet UIImageView *imageView;
@property (weak, nonatomic) IBOutlet UIActivityIndicatorView *activity;
@end

@implementation PhotoViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self.activity startAnimating];

    self.navigationItem.title = self.imageTitle;
}

- (void)viewDidAppear:(BOOL)animated;
{
    [super viewDidAppear:animated];
    
    
    AppDelegate* appDelegate = [AppDelegate delegate];
    [appDelegate viewController:self beginBackgroundDownloadForURL:self.url];
    
}

-(void)viewWillDisappear:(BOOL)animated;
{
    [super viewWillDisappear:animated];
    [self setImage:nil];
}

-(void)setImage:(UIImage *)image;
{
    [self.activity stopAnimating];
    self.imageView.image = image;
}

@end

File: General/MilleMerciTestSwift/MilleMerciTestSwift/PhotoViewController.swift
//
//  PhotoViewController.swift
//  MilleMerciTestSwift
//
//  Created by Ancil on 10/3/15.
//  Copyright © 2015 Ancil Marshall. All rights reserved.
//

import UIKit

class PhotoViewController: UIViewController {

    var photoData:PhotoData?
    let networkManager = NetworkManager()
    
    @IBOutlet weak var photoImageView: UIImageView!
    @IBOutlet weak var activity: UIActivityIndicatorView!
    

    override func viewWillAppear(animated: Bool) {
        super.viewWillAppear(animated)
        
        activity.startAnimating()
        
        if let photoData = photoData {
            navigationItem.title = photoData.title
            networkManager.beginBackgroundDownload(photoData.url) { (imageData) -> Void in
            
                let image = UIImage(data: imageData)!
                
                dispatch_async(dispatch_get_main_queue()) {
                    self.activity.stopAnimating()
                    self.photoImageView.image = image;
                }
            }
        }
    }
    
    override func viewWillDisappear(animated: Bool) {
        super.viewWillDisappear(animated)
        dispatch_async(dispatch_get_main_queue()) {
            self.activity.stopAnimating()
            self.photoImageView.image = nil;
        }
    }


}
